expression.go

```diff
@@ -173,15 +111,22 @@ func (expr FieldExpr) String() string {
 	}
 }
 
-func (expr FieldExpr) RootRefs() []FieldExpr {
-	if expr.Root != 0 {
-		return nil
+type FuncExpr struct {
+	FuncName string
+	Params   []Expression
+}
+
+func (expr FuncExpr) String() string {
```

Comment:
if this method is called often, would be better to use stringbuffer to reduce memory allocation
otherwise it is not a big deal.

Reply:
These `String()` methods are strictly used for debugging to allow you to view the structure of things.  It might make sense to call them `DebugString()` or something, but then you would loose automatic conversion by the Go string formatter.